### ISSUE TITLE
Suggestion on CSS bundling behavior

### ISSUE THREAD

**Issue by alex-sherwin** [withastro/docs#5697]
### 📋 Explain your issue

When trying debug why importing CSS stylesheets from `node_modules` was resulting in inlined `<style>` tags on each HTML page instead of externalized CSS files, I poured over the Astro docs on CSS importing behavior...

The CSS imports section only talks about doing CSS `import` statements from scripts and that `<link href="..."/>` tags can **only** use absolute URLs or relative to /public.

The confusion stems from the fact that I observed CSS inlining into `<style>` by using an `import` statement from scripting, and you're not allowed to use a `<link href=".."/>` tag to reference content from `node_modules`

The information necessary to solve my issue is actually about production specific bundling behavior which has some default magic configuration for inlining CSS styles under a certain size into HTML.

But this is an implementation/bundling behavioral thing that is logically associated with how someone would research how to import CSS and how they would behave.

# tldr

I think the CSS import section should at least make mention that CSS files imported using `import` from scripts are processed by the bundler and may or may not be inlined depending on (link to production bundling behavior section)

**Comment by lilnasy** [comment]
We do have a section dedicated to that here: https://docs.astro.build/en/guides/styling/#production

The confusion may have come from the fact that there two meanings for the word "inlining":
- First is "verbatim" (`<style is:inline>`) - the final generated HTML looks exactly how you authored it. The contents are not optimized or optimized.
- Second is "inclusion" (`defineConfig({ build: { inlineStylesheets: 'always' } })`) - after optimization, the contents are included in the generated HTML, instead of being written to a separated CSS file that is only referenced in the HTML.

**Comment by sarah11918** [comment]
Thanks, we have discussed this but won't be making any documentation changes at this time!
